src/net_processing.cpp

```diff
@@ -1443,8 +1443,8 @@ static void RelayAddress(const CNode& originator,
     std::array<std::pair<uint64_t, CNode*>,2> best{{{0, nullptr}, {0, nullptr}}};
     assert(nRelayNodes <= best.size());
 
-    auto sortfunc = [&best, &hasher, nRelayNodes, &originator](CNode* pnode) {
-        if (pnode->RelayAddrsWithConn() && pnode != &originator) {
+    auto sortfunc = [&best, &hasher, nRelayNodes, &originator, &addr](CNode* pnode) {
+        if (pnode->RelayAddrsWithConn() && pnode != &originator && (addr.IsAddrV1Compatible() || pnode->m_wants_addrv2)) {
```

Comment:
Does it make sense to add a new method `CNode::SupportsAddress(const CAddress&)` and do here: `pnode->SupportsAddress(addr)` and in `CNode::PushAddress()` instead of the variable `addr_format_supported`?

Reply:
That makes sense, but if m_wants_addrv2 is going to move to net_processing's data structures (where it IMO belongs) it's perhaps wasteful to push more into net now?

Reply:
Done.